File: cds3/B_Fibonaccharsis.py
def solution(T):
	n,k=map(int,input().split())

	cnt=0
	for i in range(n,0,-1):
		b=n
		a=i
		c=0
		while True:
			# print(a,b)
			[a,b]=[b-a,a]
			if b<0:
				break
			c+=1
		if c>=k:
			cnt+=1
		# print(c)
	print(cnt)


TT=1
TT=int(input())
for T in range(1,TT+1):
	solution(T)

File: cds3/B_Fibonaccharsis.cpp
#include <iostream>
using namespace std;

void solution(int T) {
    int n, k;
    cin >> n >> k;

    int cnt = 0;
    for (int i = n; i > 0; i--) {
        int a = i;
        int b = n;
        int c = 0;
        while (true) {
            int temp = a;
            a = b - a;
            b = temp;
            if (b < 0) {
                break;
            }
            c++;
        }
        if (c >= k) {
            cnt++;
        }
    }
    cout << cnt << endl;
}

int main() {
    int TT = 1;
    cin >> TT;
    for (int T = 1; T <= TT; T++) {
        solution(T);
    }
    return 0;
}
